### PR TITLE
DAOS-6710 rsvc: get_attr to continue upon non-existent attr

### PR DESCRIPTION
**Pull request by bfaccini**
get_attr() operation exits with -DER_NONEXIST error upon first
non-existent attribute. This patch changes this behaviour by
allowing next requested attributes to be queried.
A length of 0 will be returned for a non-existing attribute, which
is valid since empty attrs are prohibited.
-DER_NONEXIST is returned if none of the queried attrs exist.
Pool and container tests have been also been modified accordingly.

### CODE REVIEW

**Review by johannlombardi — APPROVED** [review]

**Comment by ashleypittman** [comment]
I'm conscious that we've landed and reverted this one twice already, can we do a full weekly regression build of it to ensure there are no more failures?

File: src/rsvc/srv_common.c [hunk 1]
```diff
@@ -238,15 +237,25 @@ ds_rsvc_get_attr(struct ds_rsvc *svc, struct rdb_tx *tx, rdb_path_t *path,
...
-				 svc->s_name, DP_KEY(&key), DP_RC(rc));
+		if (rc == -DER_NONEXIST) {
+			/* attribute do not exist */
+			iovs[j].iov_buf_len = sizes[i];
+			iovs[j].iov_len = 0;
+			sizes[i] = 0;
+			/* fake crt_sgl_valid() */
+			iovs[j].iov_buf = (void *)(-1);
+
+			D_DEBUG(DB_ANY, "%s: failed to lookup attribute "DF_KEY": "DF_RC"\n",
```
Comment (mchaarawi) [inline src/rsvc/srv_common.c:248]: shall we change this message from failed to non-existent?
or maybe better just remove it

**Review by mchaarawi — APPROVED** [review]

**Comment by bfaccini** [comment]
> I'm conscious that we've landed and reverted this one twice already, can we do a full weekly regression build of it to ensure there are no more failures?

@ashleypittman , I agree with you but I have used "Test-tag: pr daily_regression" , do you think it is enough ? If not, what other commit tag should I use to "do a full weekly regression build" ??

**Comment by ashleypittman** [comment]
> > I'm conscious that we've landed and reverted this one twice already, can we do a full weekly regression build of it to ensure there are no more failures?
> 
> @ashleypittman , I agree with you but I have used "Test-tag: pr daily_regression" , do you think it is enough ? If not, what other commit tag should I use to "do a full weekly regression build" ??

@sylviachanoiyee any comments?  I don't actually know what failures it was that caused us to revert this.

**Comment by sylviachanoiyee** [comment]
> > > I'm conscious that we've landed and reverted this one twice already, can we do a full weekly regression build of it to ensure there are no more failures?
> > 
> > 
> > @ashleypittman , I agree with you but I have used "Test-tag: pr daily_regression" , do you think it is enough ? If not, what other commit tag should I use to "do a full weekly regression build" ??
> 
> @sylviachanoiyee any comments? I don't actually know what failures it was that caused us to revert this.

I think running both pr and daily_regression testlist give a pretty good coverage. Running a weekly testlist will take at least 14 hours  of test time. I think it is reasonable to land this, preferably by tomorrow (Friday) and check weekly results on Monday.

**Comment by sylviachanoiyee** [comment]
The commit subject needs fixing though. It is missing the component after DAOS ref. I am unsure what component this belongs to. Note to whoever landing this, please fix title.

**Pull request merged**